### PR TITLE
Log if no WidgetPod::set_layout_rect before event

### PR DESCRIPTION
**Pull request by cmyr**
This will hopefully make an easy mistake harder to make.

motivated by [this discussion on zulip](https://xi.zulipchat.com/#narrow/stream/147926-druid/topic/children.20widgets.20could.20not.20receive.20events.20from.20parent/near/193672231)

### CODE REVIEW

File: druid/src/core.rs [hunk 1]
```diff
@@ -377,7 +380,18 @@ impl<T: Data, W: Widget<T>> WidgetPod<T, W> {
...
+        let rect = match child_ctx.base_state.layout_rect {
+            Some(rect) => rect,
+            None => {
+                log::warn!(
+                    "WidgetPod received an event without having been laid out. \
+                This likely indicates a missed call to set_layout_rect."
+                );
+                Rect::ZERO
+            }
+        };
```
Comment (Finnerale) [inline druid/src/core.rs:393]: How about
```suggestion
        if child_ctx.base_state.layout_rect.is_none() {
            log::warn!(
                "WidgetPod received an event without having been laid out. \
                This likely indicates a missed call to set_layout_rect."
            );
        }
        let rect = child_ctx.base_state.layout_rect.unwrap_or_default();
```

File: druid/src/core.rs [hunk 2]
```diff
@@ -176,20 +179,20 @@ impl<T, W: Widget<T>> WidgetPod<T, W> {
     /// Intended to be called on child widget in container's `layout`
     /// implementation.
     pub fn set_layout_rect(&mut self, layout_rect: Rect) {
-        self.state.layout_rect = layout_rect;
+        self.state.layout_rect = layout_rect.into();
```
[the remark this answers is not in the document]
Comment (cmyr) [reply druid/src/core.rs:182]: fair point!

File: druid/src/core.rs (hunk 1 above)
Comment (cmyr) [reply druid/src/core.rs:393]: I don't think I have clearly prefer your version, but since you did all that typing I'm happy to take it. ;)

**Comment by cmyr** [comment]
@Finnerale good catch, I've updated so that we don't log on `WindowConnected` and `Size`, which can arrive before a layout pass 😅

**Comment by Finnerale** [comment]
@cmyr Did you push it?

**Comment by cmyr** [comment]
just now. :)

**Pull request merged**